src/webgl/p5.Camera.js

```diff
@@ -1638,6 +1638,324 @@ p5.Camera = class Camera {
+ *   // shaking the camera to the right.
+ *   cam.slerp(cam0, cam1, amt);
+ *
+ *   background(255);
+ *   // Every time the camera turns right, the cube drifts left.
+ *   box(40);
+ * }
+ * </code>
+ * </div>
+ * @alt
+ * Prepare two cameras. One camera is facing straight ahead to the cube and the other
+ * camera is in the same position and looking straight to the right.
+ * If you use a camera which interpolates these with slerp(), the facing direction
+ * of the camera will change smoothly between the front and the right.
+ *
+ * @example
+ * <div>
+ * <code>
+ * let cam, lastCam, initialCam;
+ * let countForReset = 30;
+ * // This sample uses orbitControl() to move the camera.
+ * // Double-clicking the canvas restores the camera to its initial state.
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   strokeWeight(3);
+ *
+ *   cam = createCamera(); // main camera
+ *   lastCam = createCamera(); // Camera for recording loc info before reset
+ *   initialCam = createCamera(); // Camera for recording the initial state
+ *
+ *   setCamera(cam); // set main camera
+ * }
+ *
+ * function draw() {
+ *   if (countForReset < 30) {
+ *     // if the reset count is less than 30,
+ *     // it will move closer to the original camera as it increases.
+ *     countForReset++;
+ *     cam.slerp(lastCam, initialCam, countForReset / 30);
+ *   } else {
+ *     // if the count is 30,
+ *     // you can freely move the main camera with orbitControl().
+ *     orbitControl();
+ *   }
+ *
+ *   background(255);
+ *   box(40);
+ * }
+ * // A double-click sets countForReset to 0 and initiates a reset.
+ * function doubleClicked() {
+ *   if (countForReset === 30) {
+ *     countForReset = 0;
+ *     lastCam.set(cam);
+ *   }
+ * }
+ * </code>
+ * </div>
+ * @alt
+ * There is a camera, drawing a cube. The camera can be moved freely with
+ * orbitControl(). Double-click to smoothly return the camera to its initial state.
+ * The camera cannot be moved during that time.
+ */
+  slerp(cam0, cam1, amt) {
+    // If t is 0 or 1, do not interpolate and set the argument camera.
+    if (amt === 0) {
+      this.set(cam0);
+      return;
+    } else if (amt === 1) {
+      this.set(cam1);
+      return;
+    }
+
+    // For this cameras is ortho, assume that cam0 and cam1 are also ortho
+    // and interpolate the elements of the projection matrix.
+    if (this.projMatrix.mat4[15] !== 0) {
+      this.projMatrix.mat4[0] =
+        (1 - amt) * cam0.projMatrix.mat4[0] + amt * cam1.projMatrix.mat4[0];
+      this.projMatrix.mat4[5] =
+        (1 - amt) * cam0.projMatrix.mat4[5] + amt * cam1.projMatrix.mat4[5];
+      // If the camera is active, make uPMatrix reflect changes in projMatrix.
+      if (this._isActive()) {
+        this._renderer.uPMatrix.mat4 = this.projMatrix.mat4.slice();
+      }
+    }
+
+    // Calculate the distance between eye and center for each camera.
+    const dist0 = Math.hypot(
```

Comment:
It might be worth making a p5.Vector out of the eye and center points for both cameras, which would allow us to use `.mag()`, `.sub()`, `.lerp()`, etc below